src/libsystemd/sd-netlink/netlink-util.c

```diff
@@ -31,6 +37,16 @@ int rtnl_set_link_name(sd_netlink **rtnl, int ifindex, const char *name) {
                                 ifindex);
 
         if (strv_contains(alternative_names, name)) {
+                _cleanup_(sd_device_unrefp) sd_device *device = NULL;
+                const char *attr;
+
+                r = sd_device_new_from_ifindex(&device, ifindex);
```

Comment:
Hm, instead of creating `sd_device` object here, I think it is better that this function takes an boolean argument `bool keep_old_as_alternative` or so, as at least currently, this is called only by udevd, and it already has the `sd_device` object for the interface.